### ISSUE TITLE
rancher pipeline and alert don't work behind proxy

### ISSUE THREAD

**Issue by Wing924** [rancher/rancher#15311]
**Rancher versions:**
rancher/rancher:v2.0.8
rancher/rancher-agent:v2.0.8

**Docker version: (`docker version`,`docker info` preferred)**
1.13.1

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
CentOS Linux release 7.5.1804 (Core)
3.10.0-862.11.6.el7.x86_64

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
Bare-metal

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
single node rancher

**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
Kubernetes

**Steps to Reproduce:**

* alert
  * create a slack notifier
  * click test -> OK
  * create a alert with slack notifier -> no alert
* pipeline
  * enable pipeline

**Results:**
* no alert is sent to slack
* jenkins can't download required plugins

**Possible Reason**

alert and jenkins don't follow HTTP_PROY, HTTPS_PROXY, NO_PROXY env defined in rancher/rancher container.

**Comment by gitlawr** [comment]
There's no guarantee that rancher server and user clusters always use the same proxy server. It depends on the architect, so it would be proper to let users handle that.

**Comment by Wing924** [comment]
It would be nice if UI can set proxy because edit system workload looks dangerous.

**Comment by gitlawr** [comment]
ok, let's keep this open to see if we can have settings to configure proxy for all system components as an enhancement.

**Comment by gabrielcalderon** [comment]
I'm running into the same issue. Rancher Server is working fine with HTTP/HTTPS proxy, but when pipelines get triggered the Jenkins container has no knowledge of the proxy and thus fails.

I tested by created a simple stage that pulls **docker:latest** image and it fails.

**Comment by Woklet** [comment]
@gitlawr Perhaps I'm missing something but this still doesn't seem to solve my issue. Jenkins still doesn't work with the proxy and throws out errors if I check the jenkins workload log:

`WARNING: Failed to connect to Kubernetes at https://kubernetes.default: http://{proxy address}: Name or service not known`

I've tried various values for {proxy address} including the internal k8s address for a cntlm container, a dns address that's resolvable from all other pods and a completely outside the cluster address.

@gabrielcalderon Did you manage to solve this on your side?

**Comment by gabrielcalderon** [comment]
@Woklet Nope, I even try setting the Java proxy environment variables in the Jenkins pod. I gave uo on using the pipelines features and went back to handling CI/CD from Gitlab-CI. 

I did notice the jenkins-pipeline-slave workload tries to resolve DNS using kube-dns which fails causes it tries using /etc/resolv.conf instead of using the proxy.

**Comment by AkashEMC** [comment]
I was successfully able to build through jenkins by setting http proxy, but image publish is failing for me with logs:

<span class="timestamp"><b>12:43:18</b> </span>Debug Mode (client): false
<span class="timestamp"><b>12:43:18</b> </span>Debug Mode (server): false
<span class="timestamp"><b>12:43:18</b> </span>Registry: <a href='https://index.docker.io/v1/'>https://index.docker.io/v1/</a>
<span class="timestamp"><b>12:43:18</b> </span>Labels:
<span class="timestamp"><b>12:43:18</b> </span>Experimental: false
<span class="timestamp"><b>12:43:18</b> </span>Insecure Registries:
<span class="timestamp"><b>12:43:18</b> </span> 127.0.0.0/8
<span class="timestamp"><b>12:43:18</b> </span>Live Restore Enabled: false
<span class="timestamp"><b>12:43:18</b> </span>
<span class="timestamp"><b>12:43:18</b> </span>WARNING: No swap limit support
<span class="timestamp"><b>12:43:18</b> </span>WARNING: bridge-nf-call-iptables is disabled
<span class="timestamp"><b>12:43:18</b> </span>WARNING: bridge-nf-call-ip6tables is disabled
<span class="timestamp"><b>12:43:18</b> </span>+ /usr/local/bin/docker build --rm=true -f ./Dockerfile -t 00000000 . --pull=true --label org.label-schema.build-date=2019-08-02T12:43:18Z --label org.label-schema.vcs-ref=00000000 --label org.label-schema.vcs-url=
<span class="timestamp"><b>12:43:19</b> </span>Sending build context to Docker daemon  6.641MB

<span class="timestamp"><b>12:43:19</b> </span>Step 1/5 : FROM golang:1.11
<span class="timestamp"><b>12:43:19</b> </span>Get <a href='https://registry-1.docker.io/v2/'>https://registry-1.docker.io/v2/</a>: dial tcp 34.233.151.211:443: getsockopt: connection refused
<span class="timestamp"><b>12:43:19</b> </span>time="2019-08-02T12:43:19Z" level=fatal msg="exit status 1"


Do i need to set proxy in docker-registry service as well ?

**Comment by jawadqur** [comment]
Is there any update to this issue? I can't seem to clone github repositories when behind a proxy in the build slaves and the HTTPS_PROXY env is not set there. It is there on the master

**Comment by amyles** [comment]
Same issue on Rancher 2.4.3. We have Rancher on-prem and want our pipelines to pull from GitLab.com which can't be reached w/o our corporate proxy.

**Comment by Marouan-chak** [comment]
I'm having the same issue on Rancher 2.4.5
Is there any workaround to fix that ?
Any help would be appreciated.

**Comment by wpwoodjr** [comment]
Same issue here on Rancher 2.4.4 - Docker build step (`npm ci`) fails because npm cannot get through the corporate proxy.

**Comment by RRadziejewski** [comment]
@AkashEMC  you need to rebuild image used for this with correct proxy/no_proxy settings
In my example it looks like:

FROM rancher/plugins-docker:18.09

ENV HTTP_PROXY=http://proxy:81/
ENV HTTPS_PROXY=http://proxy:81/
ENV NO_PROXY=docker-registry.p-sn8n4-pipeline,127.0.0.0/8,localhost,127.0.0.1,0.0.0.0,10.42.0.0/16,10.43.0.0/16,192.168.10.0/24,example.com


You need to add correct proxy and no proxy so it is able to communicate to registry.

**Comment by dalvarin** [comment]
Same issue on 2.5.2 and RKE HA. We are not be able to clone the github example repos if Rancher is behind a proxy. Please, could you provide a guide about how to configure the pipeline/workloads/whatever in that case?

**Comment by RRadziejewski** [comment]
Rebuild all images which are needed for pipeline and add HTTP envs and no proxy envs to dockerfiles:)
As i've written in above comment
